tools/src/test/java/org/apache/kafka/tools/consumer/group/ResetConsumerGroupOffsetTest.java

```diff
@@ -62,506 +91,787 @@
-    public void testResetOffsetsAllTopicsAllGroups() throws Exception {
-        String[] args = buildArgsForAllGroups("--all-topics", "--to-offset", "50");
-        List<String> topics = IntStream.rangeClosed(1, 3).mapToObj(i -> TOPIC + i).collect(Collectors.toList());
-        List<String> groups = IntStream.rangeClosed(1, 3).mapToObj(i -> GROUP + i).collect(Collectors.toList());
-        topics.forEach(topic -> produceMessages(topic, 100));
+    @ClusterTemplate("generator")
+    public void testResetOffsetsExistingTopicSelectedGroups(ClusterInstance cluster) throws Exception {
+        for (GroupProtocol groupProtocol : cluster.supportedGroupProtocols()) {
+            String topic = generateRandomTopic();
 
-        for (String topic : topics) {
+            produceMessages(cluster, topic, 100);
+            List<String> groups = generateIds(topic);
             for (String group : groups) {
-                ConsumerGroupExecutor executor = addConsumerGroupExecutor(3, topic, group, GroupProtocol.CLASSIC.name);
-                awaitConsumerProgress(topic, group, 100);
-                executor.shutdown();
+                try (AutoCloseable consumerGroupCloseable =
+                             consumerGroupClosable(cluster, 1, topic, group, groupProtocol)) {
+                    awaitConsumerProgress(cluster, topic, group, 100L, groupProtocol);
+                }
             }
+
+            String[] args = buildArgsForGroups(cluster, groups, "--topic", topic, "--to-offset", "50");
+            resetAndAssertOffsets(cluster, args, 50, true, singletonList(topic), groupProtocol);
+            resetAndAssertOffsets(cluster, addTo(args, "--dry-run"),
+                    50, true, singletonList(topic), groupProtocol);
+            resetAndAssertOffsets(cluster, addTo(args, "--execute"),
+                    50, false, singletonList(topic), groupProtocol);
         }
-        resetAndAssertOffsets(args, 50, true, topics);
-        resetAndAssertOffsets(addTo(args, "--dry-run"), 50, true, topics);
-        resetAndAssertOffsets(addTo(args, "--execute"), 50, false, topics);
     }
 
-    @Test
-    public void testResetOffsetsToLocalDateTime() throws Exception {
-        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS");
-        Calendar calendar = Calendar.getInstance();
-        calendar.add(Calendar.DATE, -1);
-
-        produceMessages(TOPIC, 100);
+    @ClusterTemplate("generator")
+    public void testResetOffsetsExistingTopicAllGroups(ClusterInstance cluster) throws Exception {
+        for (GroupProtocol groupProtocol : cluster.supportedGroupProtocols()) {
+            String topic = generateRandomTopic();
+            String[] args = buildArgsForAllGroups(cluster, "--topic", topic, "--to-offset", "50");
+
+            produceMessages(cluster, topic, 100);
+            for (int i = 1; i <= 3; i++) {
+                String group = GROUP_PREFIX + i;
+                try (AutoCloseable consumerGroupCloseable =
```

Comment:
@chia7712, There is a weird problem, I use the try with auto closeable resource, I should not use `onsumerGroupCloseable.close()` this to close resource. If I don't use `close()` function, when I testing Kraft and co-Kraft mode with multi groupProtocol, it will be fail. Please take a look, thank you

Reply:
Hi @m1a2st, I remove `consumerGroupCloseable.close()` and both KRAFT and CO_KRAFT can pass. Not sure whether you can use `String group = GROUP_PREFIX + groupProtocol.name() + i;` and try it again, because you use same group names for both group protocols.

Reply:
@FrankYang0529, Thanks for your comment, and I change to use random group id and then tests will pass.